app/assets/src/components/ui/controls/GeoSearchInputBox.jsx

```diff
@@ -15,11 +15,22 @@ export const LOCATION_UNRESOLVED_WARNING =
 // Process location selections and add warnings.
 export const processLocationSelection = (result, isHuman) => {
   let warning = "";
+  // For human samples, drop the city part of the name and show a warning.
+  // NOTE: The backend will redo the geosearch for confirmation and re-apply
+  // this restriction.
   if (isHuman && get("geo_level", result) === "city") {
     result = Object.assign({}, result); // make a copy to avoid side effects
-    // For human samples, drop the city part of the name and show a warning.
-    // NOTE: The backend will redo the geosearch for confirmation and re-apply
-    // this restriction.
+
+    // If the subdivision name is identical to the city name, remove the subdivision name as well to be safe.
+    if (result.subdivision_name === result.city_name) {
```

Comment:
did you have some specific cases in mind for this? did "San Francisco" get returned like that?